pkg/spdx/document.go

```diff
@@ -205,20 +205,58 @@ func (d *Document) Write(path string) error {
 	return nil
 }
 
+// RenderFormat is valid format for rendering an SPDX document.
+type RenderFormat string
+
+// RenderFormatTagValue is the default format for an SPDX document.
+const RenderFormatTagValue = "tag/value"
+
+// RenderFormatJSON is the JSON version of a tag/value-formatted SPDX document.
+const RenderFormatJSON = "json"
+
+// RenderOption is a function with operates on a renderSettings object.
+type RenderOption func(*renderSettings)
+
+type renderSettings struct {
+	format RenderFormat
+}
+
+// WithFormat returns an RenderOption setting the format.
+func WithFormat(format RenderFormat) RenderOption {
+	return func(settings *renderSettings) {
+		settings.format = format
+	}
+}
+
 // Render reders the spdx manifest
-func (d *Document) Render() (doc string, err error) {
-	var buf bytes.Buffer
-	funcMap := template.FuncMap{
-		// The name "title" is what the function will be called in the template text.
-		"dateFormat":   func(t time.Time) string { return t.UTC().Format("2006-01-02T15:04:05Z") },
-		"extDocFormat": func(ed ExternalDocumentRef) string { logrus.Infof("External doc: %s", ed.ID); return ed.String() },
+func (d *Document) Render(options ...RenderOption) (doc string, err error) {
+	settings := &renderSettings{
+		format: RenderFormatTagValue,
+	}
+	for _, option := range options {
+		option(settings)
 	}
 
 	if d.Name == "" {
 		d.Name = "SBOM-SPDX-" + uuid.New().String()
 		logrus.Warnf("Document has no name defined, automatically set to " + d.Name)
 	}
 
+	// Convert to specific format if requested
+	switch settings.format {
+	case RenderFormatJSON:
+		logrus.Info("Rendering document as JSON")
+		b, err := json.Marshal(d)
+		return string(b), err
+	}
+
```

Comment:
Any of the logic added here will not be included in the JSON ... it seems like these types of relationships should happen ahead of time instead of being printed out as strings, but I don't have a full understanding of SPDX and what that entails